aten/src/ATen/native/quantized/cudnn/Conv.cpp

```diff
@@ -332,10 +331,26 @@ at::Tensor PackedConvWeightCudnn<kSpatialDim>::apply_impl(
       output_scale,
       output_zero_point,
       at::MemoryFormat::ChannelsLast);
-  // requantization
-  // out_int8 = act_int8 * weight_int8 * act_scale * w_scale / output_scale
-  apply_impl_helper<kReluFused>(
-      quantized_output, act, output_scale);
+
+  // cudnn v8.4.0 expects conv2d's activation tensor's input channels to be a multiple of 4. if it is not
+  // we need to explicitly pad it to a multiple of 4 ourselves as cudnn does not currently support padding.
+  // TODO: when and if cudnn enables padding in their operators, we can remove padding on our end;
+  // currently, limit padding support to groups=1 (ungrouped conv)
+  // TODO: implement this for groups > 1; should be straightforward since we're only padding a single dimension
+  if (act.size(1) % 4 != 0) {
+    int8_t num_slices = 4 - act.size(1) % 4; // number of slices we need to pad
+    auto act_padded = at::pad(act, {0, 0, 0, 0, 0, num_slices, 0, 0}, "constant", 0);
+    apply_impl_helper<kReluFused>(
+        quantized_output, act_padded.to(c10::MemoryFormat::ChannelsLast), output_scale);
+  } else {
+    apply_impl_helper<kReluFused>(
+        quantized_output, act.to(c10::MemoryFormat::ChannelsLast), output_scale);
```

Comment:
we can move this call outside of the `if else block` I think, by having the branch just do padding